devlab/tests/base.py

```diff
@@ -246,10 +247,50 @@ def user_has_not_primary_tenants(self, user_name):
                 return True
         return False
 
-    def check_vm_state(self, srv):
-        srv = self.novaclient.servers.get(srv)
+    def check_vm_state(self, srv_id):
+        srv = self.novaclient.servers.get(srv_id)
         return srv.status == 'ACTIVE'
 
+    def check_image_state(self, img_id):
+        img = self.glanceclient.images.get(img_id)
+        return img.status == 'active'
+
+    def check_volume_state(self, vol_id):
+        vlm = self.cinderclient.volumes.get(vol_id)
+        if vlm.status == 'available' or vlm.status == 'in-use':
+            return True
+        elif vlm.status == 'error':
+            msg = 'Volume with id {0} was created with error'
+            raise RuntimeError(msg.format(vol_id))
+        return False
+
+    def check_snapshot_state(self, snp_id):
+        snp = self.glanceclient.images.get(snp_id)
+        if snp.status == 'active':
+            return True
+        elif snp.status == 'error':
+            msg = 'Snapshot with id {0} has become in error state'
+            raise RuntimeError(msg.format(snp_id))
+        return False
+
+    @staticmethod
+    def wait_until_objects_created(obj_list, check_func, timeout):
+        waiting = 0
+        delay = 1
+        while waiting < timeout:
+            time.sleep(delay)
```

Comment:
nooo, you did nothing and already want to wait, why? Move time.sleep to the end of this block